www/include/reporting/dashboard/csvExport/csv_ServiceLogs.php

```diff
@@ -66,73 +66,131 @@
 
 $centreon = $oreon;
 
-/*
- * getting host and service id
- */
-isset($_GET["host"]) ? $host_id =  htmlentities($_GET["host"], ENT_QUOTES, "UTF-8") : $host_id = "NULL";
-isset($_POST["host"]) ? $host_id =  htmlentities($_POST["host"], ENT_QUOTES, "UTF-8") : $host_id;
-isset($_GET["service"]) ? $service_id =  htmlentities($_GET["service"], ENT_QUOTES, "UTF-8") : $service_id = "NULL";
-isset($_POST["service"]) ? $service_id =  htmlentities($_POST["service"], ENT_QUOTES, "UTF-8") : $service_id;
+// getting host and service id
+$hostId = filter_var(
+    $_GET['host'] ?? $_POST['host'] ?? null,
+    FILTER_VALIDATE_INT
+);
 
-/*
- * Getting time interval to report
- */
+$serviceId = filter_var(
+    $_GET['service'] ?? $_POST['service'] ?? null,
+    FILTER_VALIDATE_INT
+);
+
+// finding the user's allowed resources
+$services = $centreon->user->access->getHostServiceAclConf($hostId, 'broker', null);
+
+//checking if the user has ACL rights for this resource
+if (!$centreon->user->admin
+    && $serviceId !== null
+    && (!array_key_exists($serviceId, $services))
+) {
+    echo '<div align="center" style="color:red">' .
+        '<b>You are not allowed to access this service</b></div>';
+    exit();
+}
+
+// Getting time interval to report
 $dates = getPeriodToReport();
 $start_date =  htmlentities($_GET['start'], ENT_QUOTES, "UTF-8");
 $end_date =  htmlentities($_GET['end'], ENT_QUOTES, "UTF-8");
-$host_name = getHostNameFromId($host_id);
-$service_description = getServiceDescriptionFromId($service_id);
-
-/*
- * file type setting
- */
+$host_name = getHostNameFromId($hostId);
```

Comment:
CamelCase mandatory

Reply:
Refacto the whole code is out of my scope... I already modified the variable on which I worked on ;)

Reply:
Refacto the whole code is out of my scope... I already modified the variable on which I worked on ;)